base/inference.jl

```diff
@@ -3276,44 +3276,6 @@ function replace_getfield!(ast, e::ANY, tupname, vals, sv, i0)
     end
 end
 
-function code_typed(f, types::ANY; optimize=true)
-    types = to_tuple_type(types)
-    code_typed(call, Tuple{isa(f,Type)?Type{f}:typeof(f), types.parameters...}, optimize=optimize)
```

Comment:
I another word, I think moving it out of `inference.jl` already fixes the issue you mentioned (which is the reason behind #11590).

```julia
julia> code_typed.env.defs.func.code.module
Base

julia> methods(code_typed)
# 2 methods for generic function "code_typed":
code_typed(f::Function, types::ANY) at reflection.jl:220
code_typed(f, t::ANY) at reflection.jl:241
```